src/components/Product/ProductCard/ProductCard.tsx

```diff
@@ -19,19 +20,14 @@ export default function ProductCard({
 }: Readonly<CardProps>) {
   const addToCart = useCombinedStore((state) => state.add)
   const token = useAuthStore((state) => state.token)
-  const { addFavourite, removeFavourite, favouriteIds } = useFavouritesStore()
+  const { addFavourite, removeFavourite, favourites, favouriteIds } = useFavouritesStore()
+
+  const isInFavourites = favourites?.some((fav) => fav.id === id)
+
   const isActive = favouriteIds.includes(id)
 
   const handleButtonClick = async () => {
-    try {
-      if (isActive) {
-        await removeFavourite(id, token)
-      } else {
-        await addFavourite(id, token)
-      }
-    } catch (error) {
-      console.error('Error in handleButtonClick:', error)
-    }
+    await handleFavouriteButtonClick(id, token, isInFavourites, isActive, addFavourite, removeFavourite)
```

Comment:
nit: this code seems to be repeated; not sure if it's possible to refactor, but I'd consider this in another PR

Reply:
Absolutely right! This function with check is currently used in 4 different components in our project. That's why I put it in a separate utility and just call it in the necessary components by passing the necessary properties to it. Is it better to do with one line of code in a component or to write this huge function repeatedly in each component? Or what refactor do you mean?

<img width="1010" alt="handleButtonClick" src="https://github.com/Sunagatov/Iced-Latte-Frontend/assets/82167806/3539446a-19b1-42b4-ab86-300642a09c6f">
<img width="670" alt="handleFav" src="https://github.com/Sunagatov/Iced-Latte-Frontend/assets/82167806/f0d5cf5d-50e1-4722-b3ee-08212199c957">